docs/adr/0000-database-upgrade-testing.md

```diff
@@ -0,0 +1,60 @@
+# Database Upgrade Testing
+
+* Status: accepted
+* Deciders: [Sander, Arnaud, Wouter]
+* Date: 12/01/2023
+
+## Context and Problem Statement
+
+When making updates to the database schema, it is important to test this.
```

Comment:
I think we should expand a bit on this. We don't just want to test the migrations (we already do that), we want a uniform approach to set up any relevant preconditions for the tests. Since the considered options focus on that part, I think the problem statement should as well.

Reply:
I personally care much less about uniform and more about correct. 

Could you make a suggestion?

Reply:
That makes sense. Either way, the part where some of us made mistakes against, is the precondition set up, not the testing itself. A suggestion:
```suggestion
When making updates to the database schema, it is important to test this. In case the migration includes any data
transformation, testing this commonly involves setting up some known preconditions, and verifying that they are
converted as expected when the migration script is executed. These preconditions should be set up in a way that
remains both valid and relevant for the test scenario, even when related code may change in the future.

How do we set up these preconditions so that the tests remain valid and relevant in the future?
```

And then perhaps a small addition to the decision drivers. I'll make a separate suggestion there.